src/Component.h

```diff
@@ -87,7 +87,7 @@ struct Component {
     bool hasAddonCmake;
     std::string type;
     size_t index, lowlink;
-    std::string additionalTargetParameters;
+    std::set<std::string> additionalTargetParameters;
```

Comment:
Specific order is not relevant? This will result in EXCLUDE_FROM_ALL SHARED if you entered SHARED EXCLUDE_FROM_ALL, which may change the meaning. I'd use a vector here for that reason.

Reply:
The order is not relevant here with regards of your example.